jbi/queue.py

```diff
@@ -240,11 +236,29 @@ def ready(self) -> list[dockerflow.checks.CheckMessage]:
                     id="queue.backend.ping",
                 )
             )
+        return results
 
+    async def check_readable(self) -> list[dockerflow.checks.CheckMessage]:
+        results = []
         try:
-            bugs_items = asyncio.run(self.retrieve())
-            for items in bugs_items.values():
-                asyncio.run(async_iter(items))
+            bugs = await self.retrieve()
+            bug_failed = 0
+            parse_success = 0
+
+            for bug_id, items in bugs.items():
+                try:
+                    async for item in items:
+                        parse_success += 1
+                except Exception as exc:
+                    logger.exception(exc)
+                    bug_failed += 1
+                    results.append(
+                        dockerflow.checks.Error(
+                            f"failed to parse events for bug {str(bug_id)}",
+                            hint="check that parked event files are not corrupt",
+                            id="queue.backend.read",
+                        )
+                    )
```

Comment:
instead of this whole block, an alternative would be to add a `bug_id` and `identifier` attributes to the `QueueItemRetrievalError` exception, and just do:

```py
try:
    bug_items = (await self.retrieve()).values()
    [[i async for i in items] for items in bug_items]
except QueueItemRetrievalError as exc:
    results.append(... {exc.bug_id}, {exc.identifier}... )
```

Reply:
Will implement something similar to this. If we fail to parse the file we may not have a valid identifier. But we'll definitely have a file path.

Reply:
Implemented something that I think does what we're after.